lib/exceptions.h

```diff
@@ -48,13 +48,31 @@ class CompilerBug final : public P4CExceptionBase {
     template <typename... T>
     CompilerBug(const char* format, T... args)
             : P4CExceptionBase(format, args...)
-    { message = "COMPILER BUG:\n" + message; }
+    // \e[31m prints the text in red
+    { message = "\e[31mCOMPILER BUG\e[0m:\n" + message; }
     template <typename... T>
     CompilerBug(const char* file, int line, const char* format, T... args)
             : P4CExceptionBase(format, args...)
-    { message = cstring("COMPILER BUG: ") + file + ":" + Util::toString(line) + "\n" + message; }
+    { message = cstring("In file: ") + file + ":" + Util::toString(line) + "\n" +
+                cstring("\e[31mCOMPILER BUG\e[0m: ") + message; }
 };
 
+// This class indicates an unimplemented feature in the compiler
+class CompilerUnimplemented final : public P4CExceptionBase {
```

Comment:
You probably want a JavaDoc comment.
Also, this is not yet used anywhere, perhaps you want at least one use case.